src/ray/core_worker/context.cc

```diff
@@ -172,12 +178,34 @@ ObjectIDIndexType WorkerContext::GetNextPutIndex() {
   return GetThreadContext().GetNextPutIndex();
 }
 
+void WorkerContext::MaybeInitializeJobInfo(const JobID &job_id,
+                                           const rpc::JobConfig &job_config) {
+  absl::WriterMutexLock lock(&mutex_);
+  if (current_job_id_.IsNil()) {
+    current_job_id_ = job_id;
+  }
+  if (google::protobuf::util::MessageDifferencer::Equals(job_config_,
+                                                         kDefaultJobConfig)) {
+    job_config_ = job_config;
+  }
+  RAY_CHECK(current_job_id_ == job_id);
+  RAY_CHECK(google::protobuf::util::MessageDifferencer::Equals(job_config_, job_config_));
+}
+
 int64_t WorkerContext::GetTaskDepth() const {
   absl::ReaderMutexLock lock(&mutex_);
   return task_depth_;
 }
 
-const JobID &WorkerContext::GetCurrentJobID() const { return current_job_id_; }
+JobID WorkerContext::GetCurrentJobID() const {
+  absl::ReaderMutexLock lock(&mutex_);
+  return current_job_id_;
+}
+
+rpc::JobConfig WorkerContext::GetCurrentJobConfig() const {
```

Comment:
const reference?

Reply:
accessing a reference to a state in critical section yields undefined behavior.